docs/site/components/subpage/subpage-directive.js

```diff
@@ -34,4 +35,62 @@ angular
         }
       }
     };
+  })
+
+  // A special class used in our MD files to tell the front end to syntax
+  // highlight the code blocks.
+  .directive('hljsClass', function() {
+    'use strict';
+
+    return {
+      // Concept borrowed from:
+      // http://www.bennadel.com/blog/2748-compiling-transcluded-content-in-angularjs-directives.htm
+      //
+      // <div class="hljs-class">
+      //   var hi = 'there';        <!-- We want this!
+      // </div>
+      //
+      // We want to replace that with:
+      //
+      // <div hljs language="javascript" source="var hi = 'there';"></div>
+      //
+      // To get the content we want, we have to use a higher priority than
+      // another same-named directive that uses transclude. This is our only
+      // chance to get it before it's clobbered.
+      priority: 1500.1,
+      restrict: 'C',
+      compile: function(element, attrs) {
+        attrs._contents = element.text();
+      }
+    };
+  })
+
+  .directive('hljsClass', function() {
```

Comment:
Yeah, this was the only way (out of several, several ways I tried) that I could get it to work. http://www.bennadel.com/blog/2748-compiling-transcluded-content-in-angularjs-directives.htm outlines the solution in greater detail.